lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1864,8 +1864,12 @@ protected function _validateAndCompleteOneToOneMapping(array $mapping)
     {
         $mapping = $this->_validateAndCompleteAssociationMapping($mapping);
 
-        if (isset($mapping['joinColumns']) && $mapping['joinColumns']) {
-            $mapping['isOwningSide'] = true;
+        if (isset($mapping['joinColumns']) && $mapping['joinColumns'] && ! $mapping['isOwningSide']) {
```

Comment:
I think this condition should be used on 3.0 to throw. On 2.x though, I wouldn't touch the condition and keep setting `isOwningSide`, but I would check the value of `isOwningSide` before setting it, and trigger a deprecation then.

Reply:
Totally agree, did not intend to change logic in any way. Thanks for spotting this!